simulator/autoware_auto_msgs_adapter/include/autoware_auto_msgs_adapter/adapter_base.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 The Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef AUTOWARE_AUTO_MSGS_ADAPTER__ADAPTER_BASE_HPP_
```

Comment:
Will these headers be referenced by other packages as libraries? If such usage is not intended, [it is recommended to move them to src directory as a private header](https://docs.ros.org/en/rolling/How-To-Guides/Ament-CMake-Documentation.html#adding-files-and-headers).

Reply:
Thanks for the reference, I've looked it up and updated it and more things about library and header visibility in the code: https://github.com/autowarefoundation/autoware.universe/pull/3876/commits/40ed42e9bb1d1ee8cf13ba072eae4a19f1e77047

Reply:
Since `ament_cmake_auto` automatically exports all files under the `include` directory, the private header is placed under the src directory same as the cpp file to avoid that (e.g. `system/autoware_auto_msgs_adapter/src/adapter_base.hpp`) For public headers your original code is right (`system/autoware_auto_msgs_adapter/include/autoware_auto_msgs_adapter/adapter_base.hpp`).


After move the headers under src directory, cpp files should be able to find the header without `target_include_directories`.
```
target_include_directories(${NODE_NAME}
  PRIVATE
  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
```

Reply:
I didn't know this, thanks. Updated in https://github.com/autowarefoundation/autoware.universe/pull/3876/commits/2c5a965edac9a7d40133bf187de4043cb638ff90

Reply:
It seems, I misunderstood this completely.

If I put it under `src/include`,  then I cannot reach them from the test files.

Reply:
I understand the problem. Sorry for the confusion. There are two ways to include the header files of the src directory from the test file:

1. use relative path 
```
#include "../src/header.cpp"
```

2. use target_include_directories for test target.
```
target_include_directories(${TEST_NAME} PRIVATE src)
```

Reply:
Thanks, now tests can reach them with the second option: https://github.com/autowarefoundation/autoware.universe/pull/3876/commits/6f2319ebbf6dd6578ec52e1696a59e8996733e0f